provision/acc_provision/acc_provision.py

```diff
@@ -921,12 +924,17 @@ def provision(args, apic_file, no_random):
             "prov_apic": prov_apic,
             "debug_apic": args.debug,
         },
-        "discovered": {
-            "infra_vlan": args.infra_vlan,
-        },
     }
-    if args.username:
-        config["aci_config"]["apic_login"]["username"] = args.username
+
+    infra_vlan = None
+    if args.infra_vlan:
+        try:
+            if int(args.infra_vlan) > 0 and int(args.infra_vlan) < 4096:
```

Comment:
Not urgent. But good to have this as type=int 
https://pymotw.com/2/argparse/#simple-examples

Reply:
done